File: Tests/executors/CXX/core/src/UploadMultipartExecutor.cpp
#include "UploadMultipartExecutor.h"

#include <fstream>

void UploadMultipartExecutor::doPost(framework::HTTPRequest& request, framework::HTTPResponse& response)
{
	for (const framework::Multipart& multipart : request.getMultiparts())
	{
		std::ofstream file(multipart.getName(), std::ios::binary);
		const std::string& data = multipart.getData();

		file.write(data.data(), data.size());
	}

	response.setResponseCode(framework::ResponseCodes::created);
	response.setBody("Finish uploading files");
}

DECLARE_EXECUTOR(UploadMultipartExecutor)

File: Tests/executors/CSharp/CoreExecutors/UploadMultipartExecutor.cs
using Framework;
using Framework.Utility;

public class UploadMultipartExecutor : HeavyOperationStatelessExecutor
{
	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		List<Multipart> multiparts = [.. request.GetMultiparts()];

		foreach (Multipart multipart in multiparts)
		{
			using FileStream stream = new(multipart.Name, FileMode.Create);
			using BinaryWriter writer = new(stream);

			writer.Write(multipart.Data);
		}

		response.SetResponseCode(ResponseCodes.Created);
		response.SetBody("Finish uploading file");
	}
}
